packages/registry/contracts/CannonRegistry.sol

```diff
@@ -179,6 +186,29 @@ contract CannonRegistry is EfficientStorage, OwnedUpgradableUpdated {
       revert FeeRequired(store.publishFee);
     }
 
+    _publish(_packageName, _variant, _packageTags, _packageDeployUrl, _packageMetaUrl);
+  }
+
+  function publishWithSubscription(
```

Comment:
ok so looks like we will need to make sure that we call the right function from the cli?  was hoping it would just automatically detect by sender address or whatever, but I guess this would be fine. either way, an important thing to note.